DBPE2ETests/DBPEndToEndTests.swift

```diff
@@ -0,0 +1,513 @@
+        let state: String
+    }
+
+    struct FakeBrokerReturnedUserProfile: Codable {
+        let id: Int
+        let profileUrl: String
+        let firstName: String
+        let lastName: String
+        let age: Int
+        let city: String
+        let state: String
+    }
+
+    func mockFakeBrokerUserProfile() -> FakeBrokerUserProfile {
+        return FakeBrokerUserProfile(firstName: "John", lastName: "Smith", age: 63, city: "Dallas", state: "TX")
+    }
+
+    var fakeBrokerAPIAddress: String {
+        "http://localhost:3001/api/"
+    }
+
+    func deleteAllProfilesOnFakeBroker() async {
+        let deleteProfilesURL = URL(string: fakeBrokerAPIAddress + "profiles")!
+        var deleteRequest = URLRequest(url: deleteProfilesURL)
+        deleteRequest.httpMethod = "DELETE"
+
+        let (responseData, response) = try! await URLSession.shared.data(for: deleteRequest)
+        validateFakeBrokerResponse(responseData: responseData, response: response)
+    }
+
+    func createProfileOnFakeBroker(_ profile: FakeBrokerUserProfile) async -> FakeBrokerReturnedUserProfile {
+        let url = URL(string: fakeBrokerAPIAddress + "profiles")!
+        var request = URLRequest(url: url)
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.httpMethod = "POST"
+        let encoder = JSONEncoder()
+        let data = try! encoder.encode(profile)
+        request.httpBody = data
+
+        let (responseData, response) = try! await URLSession.shared.data(for: request)
+        validateFakeBrokerResponse(responseData: responseData, response: response)
+
+        let decoder = JSONDecoder()
+        return try! decoder.decode(FakeBrokerReturnedUserProfile.self, from: responseData)
+    }
+}
+
+// MARK: - Testing helpers and utilities
+
+private extension DBPEndToEndTests {
+
+    /*
+     Used to check an Expectation continuously
+     i.e. for Expectations when we don't know exactly when they will complete
+     but don't want to have to wait unnecessarily since they may take some time
+     */
+    private func awaitFulfillment(of expectation: XCTestExpectation, withTimeout timeout: TimeInterval, whenCondition condition: @escaping () async -> Bool) async {
+        let task = Task {
+            await fulfillExpecation(expectation, whenCondition: condition)
+        }
+
+        await fulfillment(of: [expectation], timeout: timeout)
+        task.cancel()
+    }
+
+    // Helper function for the above
+    private func fulfillExpecation(_ expectation: XCTestExpectation, whenCondition condition: () async -> Bool) async {
+        while await !condition() { }
+        expectation.fulfill()
+    }
+
+    /*
+     Used instead of using assert etc directly so we get better error messages
+     in the log when they fail.
+     When we adopt Swift 6 can likely be replaced
+     */
+    private func assertCondition(withExpectationDescription description: String, condition: () -> Bool) {
```

Comment:
Nice idea. Curious if we don’t get the same thing from specifying a message in a standard `XCTAssert`, e.g. `XCTAssertTrue({condition}, “description”)`. Always thought this would work.

Reply:
Possibly, it'll depend on the logging thing. Can't remember if I tried it